shared/js/background/onboarding.es6.js

```diff
@@ -0,0 +1,117 @@
+/**
+* This is only injected on duckduckgo.com to assist with user onboarding
+*/
+
+function createOnboardingCode ({
+    isAddressBarQuery,
+    showWelcomeBanner,
+    showCounterMessaging,
+    extensionId,
+    browser
+} = {}) {
+    const params = JSON.stringify({
+        isAddressBarQuery,
+        showWelcomeBanner,
+        showCounterMessaging,
+        extensionId,
+        browser
+    })
+
+    // TODO: upgrade to `chrome.scripting.executeScript` when we upgrade to manifest v3
+    // as it allows to inject a function with _arguments_. Here we simulate that in a hacky way
+    return `(${onboarding.toString().replace('__injectedJSON__', `'${params}'`)})()`
+}
+
+function onboarding () {
+    const {
+        isAddressBarQuery,
+        showWelcomeBanner,
+        showCounterMessaging,
+        browser,
+        extensionId
+    } = JSON.parse(__injectedJSON__) // eslint-disable-line
+
+    function getDocumentStartData (cb) {
+        if (browser !== 'chrome') {
+            return cb(null)
+        }
+
+        window.postMessage({ type: 'documentStartDataRequest' }, window.location.origin)
```

Comment:
Can all of these `postMessage` go through a central function?

Also could the location be hardcoded to the one in the filter?

Reply:
Sorry not sure that I fully understand what you mean by "the one in the filter".
Note that there are 2 different things here:
- The post message calls from the `getDocumentStartData` function. This is to get data from another content script injected by the extension (unfortunately, we need to capture this data _before_ `chrome.webNavigation.onCommitted` fires so we need to inject another script declaratively from the extension manifest).
- The response from messages received from the site

Reply:
> Regarding hardcoding the location, I used window.location.origin to facilitate local dev. I think that this is safe as those scripts can only be injected to *.duckduckgo.com (per the manifest) so in practice window.location.origin is akin to *.duckduckgo.com here.

Note that the url filter will make this harder etc but as you mention we can fix this later.

Reply:
I'm not sure if there is an obvious place to document this, bit I think it's worthwhile perhaps making a few comments point to a central README or comment perhaps?

For example it wasn't immediately clear on the moving parts:
- `onFirstSearchPostExtensionInstall` is exposed (sometimes) on SERP
- We inject `chrome-dialogue.js` on document_start
- Later we message to the script injected on start to understand if the document was focused
- We merge this data into the calling of the `onFirstSearchPostExtensionInstall`  method on SERP

Perhaps `chrome-dialogue.js` should be renamed now it's changed to `serp-document-start-handler.js`

Reply:
I hard-coded location (using a constant when possible), renamed `chrome-dialogue.js` to `serp-document-start-handler.js`, and added comments to the file.
LMK if that still needs further changes or clarifications.